src/main/java/seedu/address/model/task/Task.java

```diff
@@ -53,6 +60,19 @@ public Task(Name name, Priority priority, Description description, Done done, Se
         this.done = done;
         this.tags.addAll(tags);
         this.optionalReminder = Optional.empty();
+        this.optionalRecurring = Optional.empty();
+    }
+
+    /** With done, no reminder but with recurring */
+    public Task(Name name, Priority priority, Description description, Done done, Set<Tag> tags, Optional<Recurring> optionalRecurring) {
+        requireAllNonNull(name, priority, description, tags);
+        this.name = name;
+        this.priority = priority;
```

Comment:
why not just trigger the recurring timer at the constructors? Instead of when updateFilteredTaskList is called in ModelManager?

Reply:
It would mean having access to the model in the task class or the recurring class as well